### ISSUE TITLE
Integration with Gradle's publish-plugin?

### ISSUE THREAD

**Issue by rankinc** [GradleUp/shadow#423]
Hi,

I have a Gradle plugin which I am shading successfully with `shadow:4.0.2`. I have also published my plugin successfully using `maven-publish` (by following [this example](https://imperceptiblethoughts.com/shadow/publishing/#publishing-with-maven-publish-plugin)), but when I then try to publish it using the `publish-plugin` I find that it has not used the dependencies defined in `configurations.shadow`.

The only information about this that I have been able to find on the Web is [Gradle#5931](https://github.com/gradle/gradle/issues/5931), which _really_ doesn't help :disappointed:. Is this a known issue with an easy solution please? It seems highly unlikely that I am the first person to encounter this.

Thanks,
Chris

**Comment by johnrengelman** [comment]
Hm, I'm not sure on this one. I do this in `shadow` itself, but I publish my jars to Bintray and then only publish a marker file to the Plugin portal.

**Comment by johnrengelman** [comment]
It actually kind of tie's back to this: https://github.com/gradle/gradle/issues/5930 and also this: https://github.com/johnrengelman/shadow/pull/362

**Comment by chrisr3** [comment]
But it begs the question of
> How does `publish-plugin` _expect_ you to customise a POM created when `automatedPublishing=false`?

though :disappointed:.

**Comment by johnrengelman** [comment]
agreed, but that's a Gradle problem, not a Shadow one.

**Comment by chrisr3** [comment]
True. But it was a problem that I was hoping that maybe you'd encountered already and so knew the answer to. I'm getting nothing but tumbleweed from the Gradle forums, which could have suggested that they considered it a non-issue. (E.g. "Go RTFM! You'll find it in the basement, inside the cupboard marked 'Beware of the leopard'").

**Comment by steven-aerts** [comment]
For the maven plugin, this plugin added a [workaround](https://github.com/johnrengelman/shadow/blob/master/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/ShadowApplicationPlugin.groovy#L43) which removes the artifacts when the mavenplugin is enabled.

I am wondering if this workaround should not be applied for the maven publish plugin too.

**Comment by johnrengelman** [comment]
So, after many years, I was forced to update how this project publishes itself and ran into I believe this same issue. Thanks to @JLLeitschuh, here is the workaround - https://github.com/johnrengelman/shadow/blob/master/gradle/publish.gradle#L84-L104

This still is not officially supported by Gradle yet (I asked when doing this), so it's still in their court.

**Comment by Goooler** [comment]
https://github.com/GradleUp/shadow/tree/main/src/docs/plugins
